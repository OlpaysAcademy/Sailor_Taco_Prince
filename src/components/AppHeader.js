// @flow

import React from 'react';
import AppBar from 'material-ui/AppBar';

import Logo from './Logo';

const AppHeader = ({ currentPageName }: any) =>
    <AppBar title={currentPageName} iconElementLeft={<Logo/>} />

AppHeader.propTypes = {
    currentPageName: React.PropTypes.string
};

export default AppHeader;