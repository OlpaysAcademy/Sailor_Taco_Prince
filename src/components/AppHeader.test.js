import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import AppHeader from './AppHeader';

// You can't use react-test-renderer with another library that uses react-dom, as they both try to call injectEnvironment
// https://github.com/facebook/react/issues/7601
// Apparently this will be fixed in React 15.4
jest.mock('react-dom');

it('renders correctly without current page name', () => {
    expect(renderer.create(
        <MuiThemeProvider>
            <AppHeader currentPageName="Spicy tacos" />
        </MuiThemeProvider>
    ).toJSON()).toMatchSnapshot();
});

it('renders correctly with current page name', () => {
    expect(renderer.create(
        <MuiThemeProvider>
            <AppHeader />
        </MuiThemeProvider>
    ).toJSON()).toMatchSnapshot();
});
