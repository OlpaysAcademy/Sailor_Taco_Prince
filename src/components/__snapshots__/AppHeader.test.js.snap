exports[`test renders correctly with current page name 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#00bcd4",
      "borderRadius": "0px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "flex",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "paddingLeft": 24,
      "paddingRight": 24,
      "position": "relative",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1100
    }
  }>
  <div
    style={
      Object {
        "marginLeft": -16,
        "marginRight": 8,
        "marginTop": 8,
        "muiPrepared": true
      }
    }>
    <img
      alt="Taco Prince"
      height="50px"
      src="test-file-stub"
      width="50px" />
  </div>
  <h1
    onTouchTap={[Function anonymous]}
    style={
      Object {
        "boxFlex": 1,
        "color": "#ffffff",
        "flex": "1",
        "fontSize": 24,
        "fontWeight": 400,
        "height": 64,
        "letterSpacing": 0,
        "lineHeight": "64px",
        "margin": 0,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingTop": 0,
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap"
      }
    }>
    
  </h1>
</div>
`;

exports[`test renders correctly without current page name 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#00bcd4",
      "borderRadius": "0px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "flex",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "paddingLeft": 24,
      "paddingRight": 24,
      "position": "relative",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1100
    }
  }>
  <div
    style={
      Object {
        "marginLeft": -16,
        "marginRight": 8,
        "marginTop": 8,
        "muiPrepared": true
      }
    }>
    <img
      alt="Taco Prince"
      height="50px"
      src="test-file-stub"
      width="50px" />
  </div>
  <h1
    onTouchTap={[Function anonymous]}
    style={
      Object {
        "boxFlex": 1,
        "color": "#ffffff",
        "flex": "1",
        "fontSize": 24,
        "fontWeight": 400,
        "height": 64,
        "letterSpacing": 0,
        "lineHeight": "64px",
        "margin": 0,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingTop": 0,
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap"
      }
    }>
    Spicy tacos
  </h1>
</div>
`;
